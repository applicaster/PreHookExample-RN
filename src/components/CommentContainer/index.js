import React, { Component } from 'react';
import PropTypes from 'prop-types';
import {
  Dimensions,
  StyleSheet,
  View,
} from 'react-native';
import CommentHeader from '../CommentHeader';
import Comment from '../Comment';

const screenWidth = Dimensions.get('window').width;
const styles = StyleSheet.create({
  commentContainer: {
    borderBottomColor: '#696A6B',
    borderBottomWidth: StyleSheet.hairlineWidth,
    width: screenWidth,
    overflow: 'hidden',
  },
});

class CommentContainer extends Component {
  renderCommentHeader() {
    const { source, type, user, createdAt } = this.props.comment;
    const { avatarImageUrl, name, userName } = user;
    const overlayHeaderOnMedia = (type === 'image') || (type === 'video') || (type === 'gallery');

    return (<CommentHeader
      avatarImageUrl={avatarImageUrl}
      createdAt={createdAt}
      name={name}
      overlay={overlayHeaderOnMedia}
      source={source}
      userName={userName}
    />);
  }

  renderComment() {
    return <Comment caption={this.props.comment.caption} />;
  }

  render() {
    const { textColor = '#FFFFFF' } = this.context;
    const containerSeparatorColor = { borderBottomColor: `${textColor}4D` };
    return (
      <View style={[styles.commentContainer, containerSeparatorColor]}>
        {this.renderCommentHeader()}
        {this.renderComment()}
      </View>
    );
  }
}

CommentContainer.propTypes = {
  comment: PropTypes.object,
};

CommentContainer.contextTypes = {
  textColor: PropTypes.string,
};

export default CommentContainer;
