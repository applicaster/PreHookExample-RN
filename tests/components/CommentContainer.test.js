import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';
import CommentContainer from '../../src/components/CommentContainer';
import { comment } from './testDataUtils';

describe('CommentContainer', () => {
  test('component renders correctly', () => {
    const tree = renderer.create(
      <CommentContainer comment={comment('someId', 'text')} />
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});

