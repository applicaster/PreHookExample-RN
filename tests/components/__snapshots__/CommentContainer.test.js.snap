// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CommentContainer component renders correctly 1`] = `
<View
  style={
    Array [
      Object {
        "borderBottomColor": "#696A6B",
        "borderBottomWidth": 0.5,
        "overflow": "hidden",
        "width": 750,
      },
      Object {
        "borderBottomColor": "#FFFFFF4D",
      },
    ]
  }
>
  <View
    style={
      Array [
        Object {
          "flexDirection": "row",
          "paddingHorizontal": 12,
          "paddingTop": 12,
        },
      ]
    }
  >
    <Image
      source={
        Object {
          "uri": "https://scontent.cdninstagram.com/t51.2885-19/11356956_452315591560419_1118499770_a.jpg",
        }
      }
      style={
        Object {
          "height": 40,
          "width": 40,
        }
      }
    />
    <View
      style={
        Array [
          Object {
            "flexDirection": "column",
            "height": 43,
            "justifyContent": "space-between",
            "paddingLeft": 10,
            "width": 750,
          },
          false,
        ]
      }
    >
      <View
        style={undefined}
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          disabled={false}
          ellipsizeMode="tail"
          numberOfLines={1}
          style={
            Array [
              Object {
                "fontSize": 14,
                "fontWeight": "600",
                "marginBottom": -2,
                "marginTop": -3,
              },
              Object {
                "color": undefined,
              },
            ]
          }
        >
          Test Name
        </Text>
        <Text
          accessible={true}
          allowFontScaling={true}
          disabled={false}
          ellipsizeMode="tail"
          numberOfLines={1}
          style={
            Array [
              Object {
                "fontSize": 10,
              },
              Object {
                "color": undefined,
              },
            ]
          }
        >
          @
          applicaster
        </Text>
      </View>
      <View
        style={undefined}
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          disabled={false}
          ellipsizeMode="tail"
          style={
            Array [
              Object {
                "color": "#FFFFFF",
                "fontSize": 12,
              },
              Object {
                "color": undefined,
              },
            ]
          }
        >
          August 4
        </Text>
      </View>
    </View>
  </View>
  <Text
    accessible={true}
    allowFontScaling={true}
    disabled={false}
    ellipsizeMode="tail"
    style={
      Array [
        Object {
          "fontSize": 14,
          "lineHeight": 18,
          "paddingBottom": 12,
          "paddingHorizontal": 12,
          "paddingTop": 6,
        },
        Object {
          "color": undefined,
        },
      ]
    }
  >
    Low. Tarmac T2 , CasCam, SPC LCAs.
    Thanks 
    <Text
      accessible={true}
      allowFontScaling={true}
      disabled={false}
      ellipsizeMode="tail"
      style={
        Object {
          "color": undefined,
        }
      }
    >
      @azpinstalls
    </Text>
     for the work, parts, advice and setup.
    
    <Text
      accessible={true}
      allowFontScaling={true}
      disabled={false}
      ellipsizeMode="tail"
      style={
        Object {
          "color": undefined,
        }
      }
    >
      #brz
    </Text>
     
    <Text
      accessible={true}
      allowFontScaling={true}
      disabled={false}
      ellipsizeMode="tail"
      style={
        Object {
          "color": undefined,
        }
      }
    >
      #gt86
    </Text>
     
    <Text
      accessible={true}
      allowFontScaling={true}
      disabled={false}
      ellipsizeMode="tail"
      style={
        Object {
          "color": undefined,
        }
      }
    >
      #ft86
    </Text>
    
  </Text>
</View>
`;
